protocol_connect.go

```diff
@@ -1245,6 +1249,26 @@ func (e *connectWireError) asError() *Error {
 	return err
 }
 
+func (e *connectWireError) UnmarshalJSON(data []byte) error {
+	// We want to be lenient if the JSON has an unrecognized or invalid code.
+	// So if that occurs, we leave the code unset but can still de-serialize
+	// the other fields from the input JSON.
+	var wireError struct {
+		Code    string               `json:"code"`
+		Message string               `json:"message,omitempty"`
+		Details []*connectWireDetail `json:"details,omitempty"`
```

Comment:
Do we need the `omitempty` tags here? I think they're meaningless for unmarshaling.

Reply:
Nope, my bad. I just copied these out of the `connectWireError` struct without contemplating relevant tweaks.

Reply:
Done